applications/nrf5340_audio/src/modules/sd_card.c

```diff
@@ -222,3 +230,72 @@ int sd_card_init(void)
 
 	return 0;
 }
+
+static volatile bool seg_read_started = false;
+static struct fs_file_t f_seg_read_entry;
+
+int sd_card_segment_read_open(char const *const filename, const char *path_to_file)
+{
+	char abs_path_name[PATH_MAX_LEN + 1] = SD_ROOT_PATH;
+	strcat(abs_path_name, path_to_file);
+	printk("abs path name:\t%s\n", abs_path_name);
+	int ret;
+
+	if (!sd_init_success) {
+		return -ENODEV;
+	}
+
+	if (strlen(filename) > CONFIG_FS_FATFS_MAX_LFN) {
+		LOG_ERR("Filename is too long");
+		return -FR_INVALID_NAME;
+	}
+
+	strcat(abs_path_name, filename);
+	fs_file_t_init(&f_seg_read_entry);
+
+	ret = fs_open(&f_seg_read_entry, abs_path_name, FS_O_READ);
+	if (ret) {
+		LOG_ERR("Open file failed");
+		return ret;
+	}
+
+	seg_read_started = true;
+
+	return 0;
+}
+
+int sd_card_segment_read(char *const data, size_t *size)
+{
+	int ret;
+
+	if (!seg_read_started) {
+		return -EBUSY;
+	}
+
+	ret = fs_read(&f_seg_read_entry, data, *size);
+	if (ret < 0) {
+		LOG_ERR("Read file failed");
+		return ret;
+	}
+
+	*size = ret;
+
+	return 0;
+}
+
+int sd_card_segment_read_close(void)
```

Comment:
Some renaming required. There is `sd_card_segment_read` and `sd_card_segment_read_close`. One is a subset of the other. 
I see your logic, but xxx_read_close is to verbs at the same time which can be confusing. 
To be discussed.

Reply:
Why is the word `segment` used? I thought this was reading a single file?

Reply:
I think I see after going through the entire PR. If the file is too large, you read out a part/segment of the given file?

Reply:
Suggestion: sd_card_segment_open(), sd_card_segment_read(), sd_card_segment_close()